core/src/main/java/org/apache/accumulo/core/file/rfile/RFile.java

```diff
@@ -528,6 +540,10 @@ public void close() throws IOException {
 
     public Writer(BlockFileWriter bfw, int blockSize) throws IOException {
       this(bfw, blockSize, (int) DefaultConfiguration.getInstance().getAsBytes(Property.TABLE_FILE_COMPRESSED_BLOCK_SIZE_INDEX), null, null);
+      long indexBlockSize = DefaultConfiguration.getInstance().getAsBytes(Property.TABLE_FILE_COMPRESSED_BLOCK_SIZE_INDEX);
+      if (indexBlockSize > Integer.MAX_VALUE || indexBlockSize < 0) {
```

Comment:
This appears to only be checking the default value... which is hard-coded. A check in the unit test for Property could ensure we don't make a stupid mistake for the default. I think we may have a validate method in Property.java, also, which might be used to do a quick verification that a value is good when a user sets it or when the process starts up and reads the config for the first time.

Reply:
Went with a unit test on the default configuration. The changes where I added Preconditions.checkArgument are checking the values when they are first read from the configuration.